src/server/game/Server/Packets/MiscPackets.cpp

```diff
@@ -451,3 +451,74 @@ WorldPacket const* WorldPackets::Misc::Dismount::Write()
 
     return &_worldPacket;
 }
+
+void WorldPackets::Misc::SaveCUFProfiles::Read()
+{
+    uint8 strLen;
+
+    _worldPacket >> Count;
+
+    if (Count > MAX_CUF_PROFILES)
+        return;
+
+    for (uint8 i = 0; i < Count; i++)
+    {
+        CUFProfiles[i] = new CUFProfile;
```

Comment:
I've never used std::unique_ptr, I don't really understand when memory leaks can appear. Do you have some examples?